x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityTransformIT.java

```diff
@@ -0,0 +1,196 @@
+            final ObjectPath getTransformObjPath = assertOKAndCreateObjectPath(
+                performRequestWithRemoteTransformUser(new Request("GET", "/_transform/simple-remote-transform"))
+            );
+            assertThat(getTransformObjPath.evaluate("count"), equalTo(1));
+            assertThat(getTransformObjPath.evaluate("transforms.0.id"), equalTo("simple-remote-transform"));
+
+            // Start the transform
+            assertOK(performRequestWithRemoteTransformUser(new Request("POST", "/_transform/simple-remote-transform/_start")));
+
+            // Get the stats
+            final Request transformStatsRequest = new Request("GET", "/_transform/simple-remote-transform/_stats");
+            final ObjectPath transformStatsObjPath = assertOKAndCreateObjectPath(
+                performRequestWithRemoteTransformUser(transformStatsRequest)
+            );
+            assertThat(transformStatsObjPath.evaluate("node_failures"), nullValue());
+            assertThat(transformStatsObjPath.evaluate("count"), equalTo(1));
+            assertThat(transformStatsObjPath.evaluate("transforms.0.id"), equalTo("simple-remote-transform"));
+
+            // Stop the transform and force it to complete
+            assertOK(
+                performRequestWithRemoteTransformUser(
+                    new Request("POST", "/_transform/simple-remote-transform/_stop?wait_for_completion=true&wait_for_checkpoint=true")
+                )
+            );
+
+            // Get stats again
+            final ObjectPath transformStatsObjPath2 = assertOKAndCreateObjectPath(
+                performRequestWithRemoteTransformUser(transformStatsRequest)
+            );
+            assertThat(transformStatsObjPath2.evaluate("node_failures"), nullValue());
+            assertThat(transformStatsObjPath2.evaluate("count"), equalTo(1));
+            assertThat(transformStatsObjPath2.evaluate("transforms.0.state"), equalTo("stopped"));
+            assertThat(transformStatsObjPath2.evaluate("transforms.0.checkpointing.last.checkpoint"), equalTo(1));
+
+            // Ensure transformed data is available locally
+            final ObjectPath searchObjPath = assertOKAndCreateObjectPath(
+                performRequestWithRemoteTransformUser(
+                    new Request("GET", "/simple-remote-transform/_search?sort=user&rest_total_hits_as_int=true")
+                )
+            );
+            assertThat(searchObjPath.evaluate("hits.total"), equalTo(3));
+            assertThat(searchObjPath.evaluate("hits.hits.0._index"), equalTo("simple-remote-transform"));
+            assertThat(searchObjPath.evaluate("hits.hits.0._source.avg_stars"), equalTo(3.6));
+            assertThat(searchObjPath.evaluate("hits.hits.0._source.user"), equalTo("a"));
+            assertThat(searchObjPath.evaluate("hits.hits.1._source.avg_stars"), equalTo(2.0));
+            assertThat(searchObjPath.evaluate("hits.hits.1._source.user"), equalTo("b"));
+
```

Comment:
I think we're missing:

```java
assertThat(searchObjPath.evaluate("hits.hits.2._source.avg_stars"), equalTo(4.0));
assertThat(searchObjPath.evaluate("hits.hits.2._source.user"), equalTo("c"));
```

here.

Reply:
Thanks I can add them. Again they were not there because the YAML test (on which this test is based) does not have them.